openmc/filter.py

```diff
@@ -663,6 +671,16 @@ class ParticleFilter(Filter):
         The number of filter bins
 
     """
+    def __eq__(self, other):
+        if type(self) is not type(other):
+            return False
+        elif len(self.bins) != len(other.bins):
+            return False
+        else:
+            return np.all(self.bins == other.bins)
+
+    __hash__ = Filter.__hash__
```

Comment:
If a class defines `__eq__`, the `__hash__` method is implicitly set to `None` (if it's not otherwise defined). So if you want to retain the method from a parent class, it's required to do this (see [here](https://docs.python.org/3/reference/datamodel.html#object.__hash__))

Reply:
Good to know! Thanks